contracts/src/rollup/RollupCore.sol

```diff
@@ -411,18 +413,18 @@ abstract contract RollupCore is IRollupCore, PausableUpgradeable {
                 // and cannot go backwards
                 require(afterInboxPosition >= prevInboxPosition, "ERRORED_INBOX_TOO_FEW");
             } else if (assertion.afterState.machineStatus == MachineStatus.FINISHED) {
-                // Assertions must consume exactly all inbox messages
-                // that were in the inbox at the time the previous assertion was created
-                require(
-                    afterInboxPosition == assertion.beforeStateData.configData.nextInboxPosition, "INCORRECT_INBOX_POS"
-                );
                 // Assertions that finish correctly completely consume the message
                 // Therefore their position in the message is 0
                 require(assertion.afterState.globalState.getPositionInMessage() == 0, "FINISHED_NON_ZERO_POS");
 
                 // We enforce that at least one inbox message is always consumed
                 // so the after inbox position is always strictly greater than previous
                 require(afterInboxPosition > prevInboxPosition, "INBOX_BACKWARDS");
```

Comment:
Do we want to consider the case where 1 inbox message produced enough block to overflow?

Reply:
Yeh, I think we should.

Reply:
This also implies that position in message is not necessarily 0 for a FINISHED state.